sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -879,12 +882,77 @@ case class GpuDecimalDivide(
     }
   }
 
+  def longDivide(batch: ColumnarBatch): Any = {
+    val castLhs = withResource(GpuExpressionsUtils.columnarEvalToColumn(left, batch)) { lhs =>
+      lhs.getBase.castTo(DType.create(DType.DTypeEnum.DECIMAL128, lhs.getBase.getType.getScale))
+    }
+    val retTab = withResource(castLhs) { castLhs =>
+      val castRhs = withResource(GpuExpressionsUtils.columnarEvalToColumn(right, batch)) { rhs =>
+        withResource(divByZeroFixes(rhs.getBase)) { fixed =>
+          fixed.castTo(DType.create(DType.DTypeEnum.DECIMAL128, fixed.getType.getScale))
+        }
+      }
+      withResource(castRhs) { castRhs =>
+        com.nvidia.spark.rapids.jni.DecimalUtils.divide128(castLhs, castRhs, -dataType.scale)
+      }
+    }
+    val retCol = withResource(retTab) { retTab =>
+      if (failOnError) {
+        withResource(retTab.getColumn(0).any()) { anyOverflow =>
```

Comment:
Nit: It might help readability to use some locals to help label the columns in `retTab`, e.g.:
```scala
val overflowed = retTab.getColumn(0)
val quotient = retTab.getColumn(1)
```